src/inference/include/ie/ie_iextension.h

```diff
@@ -241,12 +275,19 @@ CreateExtension(IExtension*& ext, ResponseDesc* resp) noexcept INFERENCE_ENGINE_
 #endif
 
 /**
+ * @deprecated The Inference Engine API is deprecated and will be removed in the 2024.0 release. For instructions on
+ * transitioning to the new API, please refer to https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html
  * @def IE_DEFINE_EXTENSION_CREATE_FUNCTION
  * @brief Generates extension creation function
  */
-#define IE_DEFINE_EXTENSION_CREATE_FUNCTION(ExtensionType)                        \
-    INFERENCE_EXTENSION_API(void)                                                 \
-    IE_CREATE_EXTENSION(std::shared_ptr<InferenceEngine::IExtension>& ext);       \
-    void IE_CREATE_EXTENSION(std::shared_ptr<InferenceEngine::IExtension>& ext) { \
-        ext = std::make_shared<ExtensionType>();                                  \
-    }
+#define IE_DEFINE_EXTENSION_CREATE_FUNCTION(ExtensionType)                                                           \
+    IE_SUPPRESS_DEPRECATED_START                                                                                     \
```

Comment:
Why do we need to suppress deprecation here?
Old macro with old extensions should really emit deprecation warning